staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go

```diff
@@ -183,6 +186,11 @@ func (a *Webhook) shouldCallHook(h *v1beta1.Webhook, attr admission.Attributes,
 		return nil, err
 	}
 
+	matches, err = a.objectMatcher.MatchObjectSelector(h, attr)
```

Comment:
shouldCallHook is currently called on all hooks before dispatching to any of them. we wanted to call this inside the loop in mutatingDispatcher#Dispatch so the labels we evaluate on are the ones on the objects sent to the webhook

Reply:
that may mean Dispatch needs to be given an unfiltered list of webhooks and a way to invoke the `shouldCallHook` function